### DISCUSSION TITLE
Impact analysis?

### DISCUSSION THREAD

**Discussion by Brian-Tehan** [netbox-community/netbox#13274] [Q&A] [answered]
Is there a way to run an impact analysis on a device to where you can get a list of all impacted circuits and overlays?

Thanks.

**Comment by candlerb** [comment, marked as answer]
Not built-in, but you could write a [report](https://docs.netbox.dev/en/stable/customization/reports/) or [custom script](https://docs.netbox.dev/en/stable/customization/custom-scripts/) to do this.

Custom scripts have a form for data input. So if you want to prompt for a device, and list items affected by that device, this would be the way to do it.

Reports would be useful if you want to, for example, provide a list of all overlays, and for each one list all the devices which it touches. Reports are able to return textual data such as a CSV file, which (AFAIK) custom scripts cannot.

This requires some python programming which talks to the Django ORM. In your code, you'd follow the data model: e.g. device links to interfaces, interface links to L2VPN termination, L2VPN termination links to overlay.
